compositor_render/src/renderer/common_pipeline.rs

```diff
@@ -0,0 +1,164 @@
+use wgpu::{util::DeviceExt, BindGroup, BindGroupLayout, Buffer};
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy, bytemuck::Pod, bytemuck::Zeroable)]
+pub struct Vertex {
+    pub position: [f32; 3],
+    pub texture_coords: [f32; 2],
+}
+
+impl Vertex {
+    pub const LAYOUT: wgpu::VertexBufferLayout<'static> = wgpu::VertexBufferLayout {
+        array_stride: std::mem::size_of::<Vertex>() as u64,
+        step_mode: wgpu::VertexStepMode::Vertex,
+        attributes: &wgpu::vertex_attr_array![0 => Float32x3, 1 => Float32x2],
+    };
+}
+
+pub struct RectangleRenderBuffers {
+    pub vertex: Buffer,
+    pub index: Buffer,
+}
+
+/// Vertex and index buffer that describe render area as an rectangle mapped to texture.
+impl RectangleRenderBuffers {
```

Comment:
Couldn't find a better name for the vertex and index buffer together, and as far as I can see wgpu does not have any concept named render buffer, so hopefully it won't be confusing